pkg/didcomm/protocol/introduce/service_test.go

```diff
@@ -1423,42 +1020,365 @@ func TestService_ProposalUnusualWithRequest(t *testing.T) {
 
 	wg.Add(3)
 
-	go handleIntroducer(&flow{
+	go checkAndHandle(&flow{
```

Comment:
it would be good to comment each usage of checkAndHandle, since it is now a generic func.

Reply:
Done!